analyzers/src/SonarAnalyzer.CSharp/Rules/LossOfFractionInDivision.cs

```diff
@@ -115,5 +146,34 @@ private static bool TryGetTypeFromAssignmentToFloatType(SyntaxNode division, Sem
             type = null;
             return false;
         }
+
+        private static int? DivisionArgumentIndex(ImmutableArray<ArgumentSyntax> arguments, SyntaxNode division)
+        {
+            foreach (var argument in arguments)
+            {
+                var argumentWithDivision = argument.DescendantNodesAndSelf().FirstOrDefault(x => x.Equals(division));
+                if (argumentWithDivision != null)
+                {
+                    return arguments.IndexOf(argument);
+                }
+            }
+            return null;
+        }
+
+        private static void FlattenTupleType(ITypeSymbol typeSymbol, List<ITypeSymbol> symbolList)
+        {
+            if (!typeSymbol.IsTupleType())
+            {
+                symbolList.Add(typeSymbol.GetSymbolType());
+            }
+            else
+            {
+                var types = ((INamedTypeSymbol)typeSymbol).TupleElements();
+                foreach (var type in types)
+                {
+                    FlattenTupleType(type.Type, symbolList);
+                }
+            }
+        }
```

Comment:
Should look like this:
```suggestion
        private static List<ITypeSymbol> FlattenTupleType(ITypeSymbol typeSymbol)
        {
            List<ITypeSymbol> flattenTupleTypes = new();
            CollectTupleTypes(flattenTupleTypes, typeSymbol);
            return flattenTupleTypes;

            static void CollectTupleTypes(List<ITypeSymbol> symbolList, ITypeSymbol typeSymbol)
            {
                if (typeSymbol.IsTupleType())
                {
                    var types = ((INamedTypeSymbol)typeSymbol).TupleElements();
                    foreach (var type in types)
                    {
                        CollectTupleTypes(symbolList, type.Type);
                    }
                }
                else
                {
                    symbolList.Add(typeSymbol.GetSymbolType());
                }
            }
        }
```